src/main/java/be/seeseemelk/mockbukkit/block/state/LootableContainerMock.java

```diff
@@ -0,0 +1,221 @@
+
+	private Map<UUID, Long> lootedPlayers;
+
+	/**
+	 * Constructs a new {@link LootableContainerMock} for the provided {@link Material}.
+	 *
+	 * @param material The material this state is for.
+	 */
+	protected LootableContainerMock(@NotNull Material material)
+	{
+		super(material);
+	}
+
+	/**
+	 * Constructs a new {@link LootableContainerMock} for the provided {@link Block}.
+	 *
+	 * @param block The block this state is for.
+	 */
+	protected LootableContainerMock(@NotNull Block block)
+	{
+		super(block);
+	}
+
+	/**
+	 * Constructs a new {@link ContainerMock} by cloning the data from an existing one.
+	 *
+	 * @param state The state to clone.
+	 */
+	protected LootableContainerMock(@NotNull ContainerMock state)
+	{
+		super(state);
+	}
+
+	@Override
+	public boolean isRefillEnabled()
+	{
+		return this.refillEnabled;
+	}
+
+	/**
+	 * Sets whether this container should be refilled.
+	 *
+	 * @param refillEnabled Whether this container should be refilled.
+	 */
+	public void setRefillEnabled(boolean refillEnabled)
+	{
+		this.refillEnabled = refillEnabled;
+	}
+
+	@Override
+	public boolean hasBeenFilled()
+	{
+		return getLastFilled() != -1;
+	}
+
+	@Override
+	public boolean canPlayerLoot(@NotNull UUID player)
+	{
+		// TODO Auto-generated method stub
+		throw new UnimplementedOperationException();
+	}
+
+	@Override
+	public boolean hasPlayerLooted(@NotNull UUID player)
+	{
+		return this.lootedPlayers != null && this.lootedPlayers.containsKey(player);
+	}
+
+	@Override
+	public @Nullable Long getLastLooted(@NotNull UUID player)
+	{
+		return lootedPlayers != null ? lootedPlayers.get(player) : null;
+	}
+
+	@Override
+	public boolean setHasPlayerLooted(@NotNull UUID player, boolean looted)
+	{
+		final boolean hasLooted = hasPlayerLooted(player);
+
+		if (this.lootedPlayers == null)
+		{
+			this.lootedPlayers = new HashMap<>();
+		}
+
+		if (hasLooted != looted)
+		{
+			if (looted)
+			{
+				this.lootedPlayers.computeIfAbsent(player, p -> System.currentTimeMillis());
+			}
+			else
+			{
+				this.lootedPlayers.remove(player);
+			}
+		}
+		return hasLooted;
+	}
+
+	@Override
+	public boolean hasPendingRefill()
+	{
+		return nextRefill != -1 && nextRefill > lastFilled;
+	}
+
+	@Override
+	public long getLastFilled()
+	{
+		return this.lastFilled;
+	}
+
+	@Override
+	public long getNextRefill()
+	{
+		return this.nextRefill;
+	}
+
+
+	/**
+	 * Sets the next refill time. This uses the Bukkit Scheduler,
+	 * therefore you have to tick it with {@link BukkitSchedulerMock#performTicks(long)}.
+	 * @param refillAt timestamp in milliseconds. -1 to clear next refill
+	 * @return The previous refill time
+	 */
+	@Override
+	public long setNextRefill(long refillAt)
+	{
+		final long oldRefill = this.nextRefill;
+
+		if (refillAt < -1)
+		{
+			refillAt = -1;
+		}
+		this.nextRefill = refillAt;
+
+		new BukkitRunnable()
+		{
+			@Override
+			public void run()
+			{
+				if (nextRefill == MockBukkit.getMock().getScheduler().getCurrentTick())
```

Comment:
As I wrote in https://github.com/MockBukkit/MockBukkit/pull/866#issuecomment-1773794893 , the paper implementation for nextRefill is in relation to the milliseconds after a specific date, not the amount of ticks the server has performed. One possible alternative to fix this is to something like this
```suggestion
if (nextRefill <= System.currentTimeMillis() + MockBukkit.getMock().getScheduler().getCurrentTick()*50)
```
This implementation is flawed as well, as it technically requires the system to run the test under a certain time. The most reasonable approach is to take this in to relation to the current system time at the moment the property nextRefill was most recently assigned